src/backend/base/langflow/graph/graph/base.py

```diff
@@ -1684,18 +1697,19 @@ def layered_topological_sort(
                     if neighbor not in vertices_ids:
                         continue
 
-                    self.in_degree_map[neighbor] -= 1  # 'remove' edge
-                    if self.in_degree_map[neighbor] == 0 and neighbor not in visited:
+                    modifiable_in_degree_map[neighbor] -= 1  # 'remove' edge
+                    if modifiable_in_degree_map[neighbor] == 0 and neighbor not in visited:
                         queue.append(neighbor)
 
                     # if > 0 it might mean not all predecessors have added to the queue
                     # so we should process the neighbors predecessors
-                    elif self.in_degree_map[neighbor] > 0:
+                    elif modifiable_in_degree_map[neighbor] > 0:
                         for predecessor in self.predecessor_map[neighbor]:
                             if predecessor not in queue and predecessor not in visited:
                                 queue.append(predecessor)
 
             current_layer += 1  # Next layer
+        logger.debug(f"before refine {str(layers)}")
```

Comment:
prob remove some of these debugs or make them more descriptive if you want to keep